src/Microsoft.DncEng.SecretManager/Commands/ValidateAllCommand.cs

```diff
@@ -0,0 +1,115 @@
+using Microsoft.DncEng.CommandLineLib;
+using Microsoft.DncEng.SecretManager.StorageTypes;
+using Mono.Options;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Command = Microsoft.DncEng.CommandLineLib.Command;
+
+namespace Microsoft.DncEng.SecretManager.Commands
+{
+    [Command("validate-all")]
+    public class ValidateAllCommand : Command
+    {
+        private readonly IConsole _console;
+        private readonly SettingsFileValidator _settingsFileValidator;
+        private readonly StorageLocationTypeRegistry _storageLocationTypeRegistry;
+        private readonly List<string> _manifestFiles = new List<string>();
+        private string _basePath;
+
+        public ValidateAllCommand(IConsole console, SettingsFileValidator settingsFileValidator, StorageLocationTypeRegistry storageLocationTypeRegistry)
+        {
+            _console = console;
+            _settingsFileValidator = settingsFileValidator;
+            _storageLocationTypeRegistry = storageLocationTypeRegistry;
+        }
+
+        public override OptionSet GetOptions()
+        {
+            return new OptionSet
+            {
+                {"m|manifest-file=", "A secret manifest file. Can be specified more than once.", m => _manifestFiles.Add(m)},
+                {"base-path=", "The base path to search for settings files.", b => _basePath = b},
+            };
+        }
+
+        public override bool AreRequiredOptionsSet()
+        {
+            return !string.IsNullOrEmpty(_basePath) || _manifestFiles.Any();
+        }
+
+        public override async Task RunAsync(CancellationToken cancellationToken)
+        {
+            bool haveErrors = false;
+            var manifestFiles = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+            foreach (string manifestFile in _manifestFiles)
+            {
+                SecretManifest manifest = SecretManifest.Read(manifestFile);
+                StorageLocationType storageType = _storageLocationTypeRegistry.Get(manifest.StorageLocation.Type);
+                if (!(storageType is AzureKeyVault azureKeyVaultStorageType))
+                {
+                    _console.WriteImportant($"Skipping non-azure-key-vault manifest {manifestFile}", ConsoleColor.Yellow);
+                    continue;
+                }
+
+                string vaultUri = azureKeyVaultStorageType.GetAzureKeyVaultUri(ParameterConverter.ConvertParameters<AzureKeyVaultParameters>(manifest.StorageLocation.Parameters));
+
+                manifestFiles.Add(vaultUri, manifestFile);
+            }
+
+
+            var settingsFiles = new List<(FileInfo environmentFile, FileInfo baseFile)>();
+            foreach (string jsonFile in Directory.EnumerateFiles(_basePath, "settings.json", SearchOption.AllDirectories))
+            {
+                var baseFile = new FileInfo(jsonFile);
+                foreach (string envFile in Directory.EnumerateFiles(baseFile.DirectoryName, "settings.*.json"))
```

Comment:
I wonder if the "settings.*.json" should be a parameter, in case someone is using a different pattern? (Though it having a default is fine too).  Or is this something we are enforcing?  Like I think dotnet.status.web has them called "appconfig" (because that's what asp.net new called them)

Reply:
Also, does "settings.json" match this pattern, since it doesn't have two dots in it? I know windows globbing is sort of magic so it might, despite the fact that it doesn't look like it should.

Reply:
The pattern does not match settings.json.

I feel we should probably enforce using settings.json. There isn't any real benefit to being different. If you feel strongly the other way though I can make it a parameter.

Reply:
It's not so much that I feel strongly, rather than there are existing pieces of code that don't currently follow the pattern that, if we don't make it parameterizable, will have to be changed (which I'm fine with, we'll just need to make those changes).

I think it's _just_ the status site that lives in this repo-that's different.

Reply:
Projects that don't follow this pattern aren't unvalidatable, they just have to manually run the validate command instead of validate-all catching them. Manually running validate is not really different than parameterizing what validate-all searches. I think the validate-all command is "validate all the settings that follow the default pattern." For special snowflake projects we can add code for them specifically. We either have to change them to use the default, or write special code for them in either case.